bench/cljam/io/bam/encoder_bench.clj

```diff
@@ -0,0 +1,31 @@
+(ns cljam.io.bam.encoder-bench
+  (:require [libra.bench :refer [defbench are]]
+            [libra.criterium :as c]
+            [cljam.test-common :as tcommon]
+            [cljam.io.sam :as sam]))
+
+(defbench encode-alignment-short-bench
+  (tcommon/with-before-after {:before (tcommon/prepare-cache!)
+                              :after (tcommon/clean-cache!)}
+    (are [f]
+         (with-open
+          [r (sam/reader f)
+           w (sam/writer (str tcommon/temp-dir "/encode-alignment-bench.bam"))]
+           (let [header (sam/read-header r)
+                 aln (sam/read-alignments r)]
+             (c/quick-bench (sam/write-alignments w aln header))))
```

Comment:
- write header and refs before writing alignments
- prefer `cljam.util/with-temp-dir` which is a dedicated macro for managing temp dirs

```suggestion
  (are [f]
       (util/with-temp-dir [d "encode-alignment-short-bench"]
         (with-open [r (sam/reader f)
                     w (sam/writer (cio/file d "out.bam"))]
           (let [header (sam/read-header r)
                 xs (vec (sam/read-alignments r))]
             (sam/write-header w header)
             (sam/write-refs w header)
             (doseq [x xs o (:options x)] o)
             (c/quick-bench
              (sam/write-alignments w xs header)))))
```